test/src/org/omegat/util/StringUtilTest.java

```diff
@@ -406,4 +408,17 @@ public void testCaseConversion() {
         assertEquals("Istanbul", StringUtil.replaceCase("\\uistanbul", Locale.ENGLISH)); // English version
         assertEquals("\u0130stanbul", StringUtil.replaceCase("\\uistanbul", new Locale("tr"))); // Turkish version
     }
+
+    @Test
+    public void testParseMagicComment() {
+        // TODO: junit5 parameterized test.
+        List<String> cases = Arrays.asList(
+                "# -*- coding: UTF-8 -*-",
+                "# comment -*- coding: UTF-8 -*-"
```

Comment:
Could we get a case where `coding: UTF-8` is just one of multiple entries? And where there is a trailing `;`?

- `-*- coding: UTF-8; foo: bar -*-`
- `-*- foo: bar; coding: UTF-8 -*-`
- `-*- coding: UTF-8; -*-`

It would also be good to have some tests for examples that *don't* parse. For instance:
- Missing the first `-*-`
- Missing the final `-*-` (?)
- ?

Reply:
I added some new test cases.